middlewares/prometheus.go

```diff
@@ -0,0 +1,65 @@
+package middlewares
+
+import (
+	"github.com/containous/traefik/types"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net/http"
+)
+
+const (
+	reqsName    = "requests_total"
+	latencyName = "request_duration_milliseconds"
+)
+
+// Prometheus is an Implementation for Metrics that exposes prometheus metrics for the number of requests,
+// the latency and the response size, partitioned by status code and method.
+type Prometheus struct {
+	reqsCounter      metrics.Counter
+	latencyHistogram metrics.Histogram
+}
+
+func (p *Prometheus) getReqsCounter() metrics.Counter {
+	return p.reqsCounter
+}
+
+func (p *Prometheus) getLatencyHistogram() metrics.Histogram {
+	return p.latencyHistogram
+}
+
+// NewPrometheus returns a new prometheus Metrics implementation.
+func NewPrometheus(name string, config *types.Prometheus) *Prometheus {
+	var m Prometheus
+	m.reqsCounter = prometheus.NewCounterFrom(
```

Comment:
This is already covered by the latency histogram's _count timeseries.

It's recommend not to split out latency results by success/failure/status code as that's very commonly misinterpreted. What I'd suggest is removing the code label from the latency metric.

It's also unwise on cardinality grounds to have too many labels on a histogram, as it already has one itself.